pkg/proxy/iptables/proxier.go

```diff
@@ -920,6 +920,11 @@ func (proxier *Proxier) syncProxyRules() {
 			writeLine(proxier.natChains, utiliptables.MakeChainLine(chainName))
 		}
 	}
+	// ensure KUBE-MARK-DROP chain exist but not do any change
+	if _, err := proxier.iptables.EnsureChain(utiliptables.TableNAT, KubeMarkDropChain); err != nil {
```

Comment:
In a way this is reverting https://github.com/kubernetes/kubernetes/pull/85527, without having kube-proxy overwrite the iptable rules from kubelet. @cmluciano mentioned this fixed the issue on this environment, @aojea @uablrek can you check if this seems okay based on past discussions in https://github.com/kubernetes/kubernetes/pull/85527 and https://github.com/kubernetes/kubernetes/pull/82214

Reply:
Lgtm. The `EnsureChain` just creates the chain if it doesn't exist, if it already exists it does not return an error.

IMO it should be some rule that any program that uses a chain **must** call `EnsureChain` before use.

The (way too long) debate whenever that is right or not can go on independently, but at least things works.

Reply:
I think that this should go before Line 872, 
https://github.com/kubernetes/kubernetes/pull/94591/files#diff-d51765b83fe795b469e8a86276b12dc9R872

the comment in that line is very explicit about it
```
// Below this point we will not return until we try to write the iptables rules.
```

and there is also a loop checking that the other chains exist

Reply:
@aojea , Fixed, thanks.